server/src/storage/datastore.ts

```diff
@@ -268,11 +268,19 @@ const addEntityHelper = async (
 
   try {
     if (uniqueProperties === undefined) {
-      actor.insert(entity);
-    } else if (actor instanceof Datastore) {
-      await insertUniqueEntity(kind, entity, uniqueProperties);
-    } else if (actor instanceof Transaction) {
-      await uniqueInsertInTransaction(kind, entity, uniqueProperties)(actor);
+      if (actor instanceof Datastore) {
+        await actor.insert(entity);
+      } else if (actor instanceof Transaction) {
+        // Do not await here because promise is resolved only after
+        // `await transaction.commit()` is called.
+        actor.insert(entity);
+      }
+    } else {
+      if (actor instanceof Datastore) {
+        await insertUniqueEntity(kind, entity, uniqueProperties);
+      } else if (actor instanceof Transaction) {
+        await uniqueInsertInTransaction(kind, entity, uniqueProperties)(actor);
+      }
```

Comment:
now we really have different impl for transaction and datastore, then there is no need to create actor var.

suggest to change to sth. like this:

if (transaction != null) {
  ...
} else {
  ...
}

Reply:
+1

Reply:
@ifhuang @chshda Okay! Will do this. I didn't realise that datastore and transaction were separate at first and only combined in the actor var.